### PR TITLE
Add more dependency versions to plugin metadata report

### PR DESCRIPTION
**Pull request by superm1**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -201,6 +201,9 @@ if get_option('plugin_uefi')
     conf.set('HAVE_FWUP_GET_BGRT_INFO', '1')
   endif
   efivar = dependency('efivar')
+  if fwup.version().version_compare('< 11')
+    conf.set('LIBFWUP_LIBRARY_VERSION', fwup.version())
```
Comment (hughsie) [inline meson.build]: can't we just use `conf.set_quoted()` and avoid the printf?

File: plugins/uefi/fu-plugin-uefi.c [hunk 2]
```diff
@@ -29,6 +29,12 @@
 #include "fu-plugin.h"
 #include "fu-plugin-vfuncs.h"
 
+/* drop when upgrading minimum required version of fwupdate to 11 */
+#if defined(LIBFWUP_LIBRARY_VERSION)
+#undef  LIBFWUP_VERSION
+#define LIBFWUP_VERSION (LIBFWUP_LIBRARY_VERSION)
```
Comment (hughsie) [inline plugins/uefi/fu-plugin-uefi.c]: Do we care about LIBFWUP_VERSION at all? Can't we just rely on pkgconfig?

File: meson.build (hunk 1 above)
Comment (superm1) [reply meson.build]: Not if we want to keep the same format as is used in fwupdate upstream currently: https://github.com/rhboot/fwupdate/blob/master/linux/include/fwup-version.h.in

File: plugins/uefi/fu-plugin-uefi.c (hunk 2 above)
Comment (superm1) [reply plugins/uefi/fu-plugin-uefi.c]: That's a good point.  I guess then can use `conf.set_quoted()` too.

Comment (superm1) [reply plugins/uefi/fu-plugin-uefi.c]: Fixed

File: plugins/dell/fu-plugin-dell.c [hunk 3]
```diff
@@ -1043,11 +1043,15 @@ void
...
 	FuPluginData *data = fu_plugin_alloc_data (plugin, sizeof (FuPluginData));
+	g_autofree gchar *tmp = NULL;
 
 	data->libsmbios_major = smbios_get_library_version_major();
 	data->libsmbios_minor = smbios_get_library_version_minor();
 	g_debug ("Using libsmbios %u.%u", data->libsmbios_major,
 		 data->libsmbios_minor);
+	tmp = g_strdup_printf ("%u.%u", data->libsmbios_major,
+					data->libsmbios_minor);
+	fu_plugin_add_report_metadata(plugin, "LibsmbiosVersion", tmp);
```
Comment (hughsie) [inline plugins/dell/fu-plugin-dell.c]: space after the bracket pls.

File: meson.build [hunk 4]
```diff
@@ -201,6 +201,9 @@ if get_option('plugin_uefi')
     conf.set('HAVE_FWUP_GET_BGRT_INFO', '1')
   endif
   efivar = dependency('efivar')
+  if fwup.version().version_compare('< 11')
```
[the remark this answers is not in the document]
Comment (superm1) [reply meson.build]: fixed

File: plugins/dell/fu-plugin-dell.c (hunk 3 above)
Comment (superm1) [reply plugins/dell/fu-plugin-dell.c]: fixed

**Pull request merged**